example/webGL/index.html

```diff
@@ -0,0 +1,206 @@
+
+				var info = document.createElement( 'div' );
+				info.style.position = 'absolute';
+				info.style.top = '10px';
+				info.style.width = '100%';
+				info.style.textAlign = 'center';
+				info.innerHTML = 'WebGL Drawing Example<br/>';
+				container.appendChild( info );
+
+				scene = new THREE.Scene();
+
+				camera = new THREE.PerspectiveCamera( 50, window.innerWidth / window.innerHeight, 1, 1000 );
+				camera.position.set( 0, 0, 300 );
+				scene.add( camera );
+
+				var light = new THREE.PointLight( 0xffffff, 0.8 );
+				camera.add( light );
+
+				group = new THREE.Group();
+				group.position.y = 50;
+				scene.add( group );
+
+				var loader = new THREE.TextureLoader();
+				var texture = loader.load( "textures/UV_Grid_Sm.jpg" );
+
+				// it's necessary to apply these settings in order to correctly display the texture on a shape geometry
+
+				texture.wrapS = texture.wrapT = THREE.RepeatWrapping;
+				texture.repeat.set( 0.008, 0.008 );
+
+				function addShape( shape, extrudeSettings, color, x, y, z, rx, ry, rz, s ) {
+
+					// lines
+					shape.autoClose = true;
+					var points = shape.createPointsGeometry();
+					var spacedPoints = shape.createSpacedPointsGeometry( 50 );
+
+					// solid line
+					var line = new THREE.Line( points, new THREE.LineBasicMaterial( { color: color, linewidth: 3 } ) );
+					line.position.set( x, y, z - 25 );
+					line.rotation.set( rx, ry, rz );
+					line.scale.set( s, s, s );
+					group.add( line );
+				}
+
+				// Triangle
+
+//				var triangleShape = new THREE.Shape();
+//				triangleShape.moveTo(  80, 20 );
+//				triangleShape.lineTo(  40, 80 );
+//				triangleShape.lineTo( 120, 80 );
+//				triangleShape.lineTo(  80, 20 ); // close path
+
+				// Square
+
+				var sqLength = 30;
+				var squareShape = new THREE.Shape();
+				squareShape.moveTo( 0,0 );
+				squareShape.lineTo( 0, sqLength );
+				squareShape.lineTo( sqLength, sqLength );
+				squareShape.lineTo( sqLength, 0 );
+				squareShape.lineTo( 0, 0 );
+
+				// Circle
+
+//				var circleRadius = 20;
+//				var circleShape = new THREE.Shape();
+//				circleShape.moveTo( 0, circleRadius );
+//				circleShape.quadraticCurveTo( circleRadius, circleRadius, circleRadius, 0 );
+//				circleShape.quadraticCurveTo( circleRadius, -circleRadius, 0, -circleRadius );
+//				circleShape.quadraticCurveTo( -circleRadius, -circleRadius, -circleRadius, 0 );
+//				circleShape.quadraticCurveTo( -circleRadius, circleRadius, 0, circleRadius );
+
+				var extrudeSettings = { amount: 8, bevelEnabled: true, bevelSegments: 2, steps: 2, bevelSize: 1, bevelThickness: 1 };
+
+//				 addShape( shape, color, x, y, z, rx, ry,rz, s );
+
+                var numObjects = 200;
+                    for (var ii = 0; ii < numObjects; ++ii) {
+//                        addShape( triangleShape, extrudeSettings, 0x8080f0, -180, 0, 0, 0, 0, 0, 1 );
+                        addShape( squareShape, extrudeSettings, 0x000000, -300+ii*30, 20, 0, 0, 0, 0, 1 );
```

Comment:
Use d3.gridding here instead of iterating over numObjects, and draw the actual x, y, h, w generated by the lib